clang/bindings/python/clang/cindex.py

```diff
@@ -2318,25 +2356,25 @@ def kind(self):
         """Return the kind of this type."""
         return TypeKind.from_id(self._kind_id)
 
-    def argument_types(self):
+    def argument_types(self) -> NoSliceSequence[Type]:
         """Retrieve a container for the non-variadic arguments for this type.
 
         The returned object is iterable and indexable. Each item in the
         container is a Type instance.
         """
 
-        class ArgumentsIterator(collections.abc.Sequence):
```

Comment:
This is a type-error since `ArgumentsIterator` doesn't actually implement the `Sequence` protocol: it does not support slicing.